src/main/java/htsjdk/variant/utils/SAMSequenceDictionaryExtractor.java

```diff
@@ -75,30 +85,53 @@ SAMSequenceDictionary extractDictionary(final File dictionary) {
                     CloserUtil.close(bufferedLineReader);
                 }
             }
+
+            @Override
+            SAMSequenceDictionary extractDictionary(Path dictionary) {
+                return extractDictionary(dictionary.toFile());
+            }
         },
         SAM(IOUtil.SAM_FILE_EXTENSION, BamFileIoUtils.BAM_FILE_EXTENSION) {
             @Override
             SAMSequenceDictionary extractDictionary(final File sam) {
                 return SamReaderFactory.makeDefault().getFileHeader(sam).getSequenceDictionary();
             }
+
+            @Override
+            SAMSequenceDictionary extractDictionary(Path sam) {
+                return SamReaderFactory.makeDefault().getFileHeader(sam).getSequenceDictionary();
+            }
         },
         VCF(IOUtil.VCF_EXTENSIONS) {
             @Override
             SAMSequenceDictionary extractDictionary(final File vcf) {
-                VCFFileReader vcfFileReader = null;
+                VCFPathReader vcfReader = null;
```

Comment:
Personally I don't think renaming the class is necessary. Even though the API now accepts/requires `Path` it is still reading a VCF File. The implementation detail of how a file is represented has changed but the functionality hasn't.

Reply:
done.